docs/spec/SPEC-SPEC.md

```diff
@@ -0,0 +1,41 @@
+# Specification of Specifications
+
+This file intends to outline the common structure for specifications within
+this directory. 
+
+
+## Common Layout
+
+The following generalized structure should be used to breakdown specifications
+for modules. Note that not all files may be required depending on the modules 
+function
+
+ - `overview.md` - describe module
+ - `state.md` - specify and describe structures expected to marshalled into the store, and their keys
+ - `hooks.md` - describe available hooks to be called by/from this module 
+ - `end_block.md` - specify any end-block operations
+ - `begin_block.md` - specify any begin-block operations
+ - `tags.md` - list and describe event tags used
+ - `transactions.md` - specify transaction structure and expected state machine behaviour 
```

Comment:
Shouldn't this really be `msgs.md`, since we're always describing `sdk.Msg` implementations?

Reply:
sure - `messages.md` seems good